src/main/cpp/src/decimal_utils.cu

```diff
@@ -520,6 +519,72 @@ inline __device__ int precision10(chunked256 value) {
     return -1;
 }
 
+__device__ chunked256 set_scale_and_round(chunked256 data, int old_scale, int new_scale) {
+    if (old_scale != new_scale) {
+        if (new_scale < old_scale) {
+            int raise = old_scale - new_scale;
+            int multiplier = pow_ten(raise).as_128_bits();
+            data = multiply(data, chunked256(multiplier));
+        } else {
+            int drop = new_scale - old_scale;
+            auto const diviser = pow_ten(drop).as_128_bits();
+            data = divide_and_round(data, chunked256(diviser).as_128_bits());
```

Comment:
Why are we converting diviser from __int128_t to chunked256 and back to __int128_t?